ckanext/xloader/utils.py

```diff
@@ -41,9 +46,72 @@ def is_it_an_xloader_format(cls, format_):
         return format_.lower() in cls._formats
 
 
+def awaiting_validation(res_dict):
+    """
+    Checks the existence of a logic action from the ckanext-validation
+    plugin, thus supporting any extending of the Validation Plugin class.
+
+    Checks ckanext.xloader.validation.requires_successful_report config
+    option value.
+
+    Checks ckanext.xloader.validation.enforce_schema config
+    option value. Then checks the Resource's validation_status.
+    """
+    if not p.toolkit.asbool(p.toolkit.config.get('ckanext.xloader.validation.requires_successful_report', False)):
+        # validation.requires_successful_report is turned off, return right away
+        return False
+
+    try:
+        # check for one of the main actions from ckanext-validation
+        # in the case that users extend the Validation plugin class
+        # and rename the plugin entry-point.
+        p.toolkit.get_action('resource_validation_show')
+        is_validation_plugin_loaded = True
+    except KeyError:
+        is_validation_plugin_loaded = False
+
+    if not is_validation_plugin_loaded:
+        # the validation plugin is not loaded but required, log a warning
+        log.warning('ckanext.xloader.validation.requires_successful_report requires the ckanext-validation plugin to be activated.')
+        return False
+
+    if p.toolkit.asbool(config.get('ckanext.xloader.validation.enforce_schema', True)):
+        # validation.enforce_schema is turned on, explicitly check for the `validation_status`
+        if res_dict.get('validation_status', None) != 'success':
+            return True
+        else:
+            return False
+
+    elif res_dict.get('schema', None) and res_dict.get('validation_status', None) != 'success':
```

Comment:
This could be combined with line 78, making the condition "if a schema exists or config strictly requires one"

Reply:
Nice, yeah that makes a lot more sense thanks!